core/src/models/Source.ts

```diff
@@ -283,7 +284,7 @@ export class Source extends LoggedModel<Source> {
     const options = await this.getOptions();
     const mapping = await MappingHelper.getMapping(this, "id");
 
-    return {
+    let configObject: any = {
```

Comment:
Perhaps type `ConfigurationObject` might be a little tighter than `any`?  Or, was the `any` added to allow you to change its makeup below with the merge?

Reply:
Yep. The merge thing.

It also wouldn't let me do something like `ConfigurationObject | ConfigurationObject[]` because they don't share properties in common with one another.